### ISSUE TITLE
[Bug]: UnixDomainSocket NullReferenceException in FramingModelHandshakeMiddleware

### ISSUE THREAD

**Issue by Arzbergerr** [CoreWCF/CoreWCF#1505]
### Duplicate ?

- [X] I have searched issues/discussions and did not find other issues/discussions reporting this bug.

### Product version

1.6.0

### Describe expected behavior

No Exception to be thrown / No EventLog entry to be created

### Describe actual behavior

Since i had not the correct options to select: first some corrections about information:
**Binding:** UnixDomainSocketBinding
**.NET version:** .NET 8

It's something similar like in [Issue #878](https://github.com/CoreWCF/CoreWCF/issues/878) but i think not completly the same since i don't get the error with **NetTcpBinding**. 
The error occurs on try to get a `IConnectionReuseHandler` in `FramingModeHandshakeMiddleware.OnConnectedAsync` when using `UnixDomainSocketBinding` for connection.

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at CoreWCF.Channels.Framing.FramingModeHandshakeMiddleware.OnConnectedCoreAsync(FramingConnection connection)
   at CoreWCF.Channels.Framing.FramingModeHandshakeMiddleware.OnConnectedAsync(FramingConnection connection)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.KestrelConnection`1.ExecuteAsync()
```

After investigation the problem is on trying to get a `NetFramingListenOptions` from the `ConnectionFeatures`:

```
if (reuseHandler == null)
{
    var listenOptions = connection.ConnectionFeatures.Get<NetFramingListenOptions>();
    reuseHandler = listenOptions.ConnectionReuseHandler;
}
```

On setup of the `UseUnixDomainSocket` there are added `IConfigureOptions<KestrelServerOptions>` with `UnixDomainSocketFrameingOptionsSetup`. This setup will add options of the type `UnixDomainSocketListenOptions` which should be created  after calling `Listen()` from `UnixDomainSocketOptions`. e.g.:

```
builder.UseUnixDomainSocket(options =>
{
    options.Listen(uri, opt=> { });
});
```

I have found some differences to the NetTcp version. The `UnixDomainSocketListenOptions`, which will be created on `Listen()`, does not inherit from `NetFramingListenOptions` like `TcpListenOptions` does. Also the `UnixDomainSocketFramingOptionsSetup` does not add the ListenOptions to the `ConnectionContext`.
See from NetTcp:

```
builder.Use(next =>
{
    return (ConnectionContext context) =>
    {
        context.Features.Set<NetFramingListenOptions>(tcpListenOptions);
        return next(context);
    };
});
```

Therefore `NetFramingListenOptions` cannot be loaded from `FramingModeHandshakeMiddleware` which results in the exception above which is also written to EventLog (if the logging is not deactivated) which results in **many** exception entries in the EventLog!

As workaround i also added a `services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<KestrelServerOptions>, MyOwnSetup>(provider => new MyOwnSetup(provider)));`. This setup will register options to be available later on:

```
internal class MyOwnSetup: IConfigureOptions<KestrelServerOptions>
{
    private readonly IServiceProvider mServiceProvider;

    public MyOwnSetup(IServiceProvider serviceProvider)
    {
        mServiceProvider = serviceProvider;
    }

    public void Configure(KestrelServerOptions options)
    {
        options.ApplicationServices = mServiceProvider;
        var listenOptions = new MyListenOptions(mServiceProvider);

        options.ConfigureEndpointDefaults(options =>
        {
            options.Use(next =>
            {
                return (ConnectionContext context) =>
                {
                    context.Features.Set<NetFramingListenOptions>(listenOptions);
                    return next(context);
                };
            });
        });
    }
}

internal class MyListenOptions : NetFramingListenOptions
{
    public MyListenOptions (IServiceProvider serviceProvider)
    {
        ApplicationServices = serviceProvider;
    }

    public override IServiceProvider ApplicationServices { get; }
}
```

I used `ConfigureEndpointDefaults` because i was not sure if `ListenUnixSocket` is a good idea since this will also be done on the existing setup.
I'am also not sure if this is the best approach but for now it works without disable the logging. 

Long story short: i guess the correct options should also be set for `UnixDomainSocket` to prevent this exception without workarounds.

### Which binding

CustomBinding

### security

Transport

### Which .NET version

.NET 6

### Which os platform

Windows

### Stacktrace if any

```shell
System.NullReferenceException: Object reference not set to an instance of an object.
   at CoreWCF.Channels.Framing.FramingModeHandshakeMiddleware.OnConnectedCoreAsync(FramingConnection connection)
   at CoreWCF.Channels.Framing.FramingModeHandshakeMiddleware.OnConnectedAsync(FramingConnection connection)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure.KestrelConnection`1.ExecuteAsync()
```
